opslevel/resource_opslevel_team_membership.go

```diff
@@ -0,0 +1,180 @@
+package opslevel
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/opslevel/opslevel-go"
+)
+
+func resourceTeamMembership() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manage a team's user membership",
+		Create:      wrap(resourceTeamMembershipCreate),
+		Read:        wrap(resourceTeamMembershipRead),
+		Update:      wrap(resourceTeamMembershipUpdate),
+		Delete:      wrap(resourceTeamMembershipDelete),
+		Importer: &schema.ResourceImporter{
+			State: resourceTeamMembershipImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"team_id": {
```

Comment:
Nothing jumps out at my as being problematic. I believe each resource has a "canonical" alias, right? I believe in that case, the module approach would be
```terraform
resource "opslevel_team" "my_team" {
  name = "My Team"
}

resource "opslevel_team_membership" "my_team" {
  team_alias = opslevel_team.wampa.alias  # This is referring to the resource alias instead of id
  members = [
    "member@myteam.com",
  ]
}
```

If a user decided to define the alias as one of the additional aliases (not referring to the team resource), I don't anticipate this would cause any problems beyond some potential drift, but that seems unavoidable. I can implement it this way

Reply:
Made an attempt with https://github.com/OpsLevel/terraform-provider-opslevel/pull/24/commits/cfec9eaa6fc6de8f94e88deb6376cabd5239d540

Let me know if you think this aligns with your expectations 🙏

Reply:
Lets just call the field `team` as we've done `group` instead of `group_alias` on the team resource previously.

As for the deprecated field (this was something that was changed in the API a while back so that teams can have multiple aliases referring them - so we have to code around it) the end user should beable to just do:
```yaml
resource "opslevel_team" "my_team" {
  name = "My Team"
}

resource "opslevel_team_membership" "my_team" {
  team = opslevel_team.my_team.aliases[0]
  members = [
    "member@myteam.com",
  ]
}
```

That doesn't feel too bad to force the end user to do and it would show as a diff which is what you'd expect to see if you changed the aliases list on a team - especially if you shorten the list you'd want to know all the things it could have potentially changed.

Reply:
Works for me, just wanted to make sure you were on board :)

Reply:
Yes the import statement should use the alias of the team as thats a much better UX.

I'm now starting to wonder - why do we not have an array field on just on the `opslevel_team`?  What is the advantage to having this be a separate resource?

Terraform docs explicitly state that generally each resource in terraform should have a unique ID associated in your API.

IE
```yaml
resource "opslevel_team" "my_team" {
  name = "My Team"
  members = [
    "member@myteam.com",
  ]
}
```

Reply:
🤔 

To be honest I was following a resource pattern I was familiar with working with the AWS provider and I can't really motivate it more than that. Something that came to mind when I was originally working on this is that it's nice to manage fewer fields on a resource, but that's a pretty flimsy argument.